common/winobjc.versioning.common.targets

```diff
@@ -78,9 +66,47 @@
       <PackageVersion_Major>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['MAJOR'].Value)</PackageVersion_Major>
       <PackageVersion_Minor>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['MINOR'].Value)</PackageVersion_Minor>
       <PackageVersion_Patch>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['PATCH'].Value)</PackageVersion_Patch>
-      <PackageVersion_PreReleaseLabel>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['LABEL'].Value)</PackageVersion_PreReleaseLabel>
+      <PackageVersion_PreRelease>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['LABEL'].Value)</PackageVersion_PreRelease>
+    </PropertyGroup>
+
+    <!-- Common properties needed to either check the build numbner or to calculate the version -->
+    <PropertyGroup Condition="'$(PackageVersionOverride)' == ''">
+      <PackageVersion_Major>$(GitVersion_Major)</PackageVersion_Major>
+      <PackageVersion_Minor>$(GitVersion_Minor)</PackageVersion_Minor>
+      
+      <PackageVersion_Patch Condition="'$(PackageVersion_Patch)' == '' And '$(GitVersion_PreReleaseLabel)' == ''">$([System.DateTime]::Now.ToUniversalTime().ToString(yyMMdd))</PackageVersion_Patch>
+      <PackageVersion_Patch Condition="'$(PackageVersion_Patch)' == '' And '$(GitVersion_PreReleaseLabel)' != ''">$(GitVersion_Patch)</PackageVersion_Patch>
+
+      <PackageVersion_PreReleaseLabel>$(GitVersion_PreReleaseLabel)</PackageVersion_PreReleaseLabel>
+    </PropertyGroup>
+
+    <!-- Last up, use calculated values. -->
+    <PropertyGroup Condition="'$(PackageVersionOverride)' == ''">
+      <!-- 
+         Three places to get a build timestamp: 
+         1. Off of the build number for a build server build (Before its been modified to be the package version)
+         2. From a timestamp file created at the beginning of a local build (see NuprojInitializer.msbuildproj)
```

Comment:
Why can't the build server publish the  timestamp file and subsequent release defs use that timestamp file?
I'd would be nice if the build were consistent for local builds as well as builds on the build server.